src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -57,29 +67,26 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
             // safe to use within IProjectDynamicLoadComponent.LoadAsync.
             _projectGuid = await _project.UnconfiguredProject.GetProjectGuidAsync();
             _missingSetupComponentRegistrationService.RegisterProjectConfiguration(_projectGuid, _project);
-            _projectFaultHandlerService.Forget(RegisterSdkNetCoreRuntimeNeededInProjectAsync(_project), _project.UnconfiguredProject);
+            _subscription = _projectSubscriptionService.ProjectRuleSource.SourceBlock.LinkToAsyncAction(target: OnProjectChangedAsync, _project.UnconfiguredProject);
```

Comment:
Subscriptions to rule sources should specify the name of rules being subscribed to. In this case you want `ConfigurationGeneral.SchemaName`. Look for examples of subscribing to `ProjectRuleSource` where rule names are passed.

Also make sure that any fault on the block is reported via `IProjectFaultHandlerService` so we don't fail silently. Reporting faults that way causes gold bars and contributes to NFE reports in our dashboards.

The `LinkToAsyncAction` extension method does use `IProjectFaultHandlerService` today. However when you change the code you'll need to make sure that behaviour is preserved, even if you end up writing the code for that manually (i.e. creating your own action block explicitly, passing rules when linking it, and also registering the block for fault handling.)

Reply:
Thank you for the information, I wasn't aware we could subscribe to specific data (it does make sense). I made sure a fault handler was still used somewhere in the replacement call